x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/Distribution/index.tsx

```diff
@@ -134,12 +139,15 @@ export const TransactionDistribution: FunctionComponent<Props> = (
   const xMax = d3.max(buckets, d => d.x) || 0;
   const timeFormatter = getDurationFormatter(xMax);
 
-  const bucketIndex = buckets.findIndex(
-    bucket =>
-      bucket.sample != null &&
-      bucket.sample.transactionId === transactionId &&
-      bucket.sample.traceId === traceId
-  );
+  const bucketIndex = buckets.findIndex(bucket => {
+    const sampleFound = bucket.samples.findIndex(
+      sample =>
+        sample.transactionId === transactionId && sample.traceId === traceId
+    );
+    return sampleFound !== -1;
+  });
+
+  onBucketSelected(bucketIndex);
```

Comment:
Having side effects in `render` is normally a bad sign. This will call `setSelectedBucket`. Instead `bucketIndex` should be calculated outside `TransactionDistribution` and passed in. 

```tsx
<TransactionDistribution
    distribution={distributionData}
    isLoading={distributionStatus === FETCH_STATUS.LOADING}
    urlParams={urlParams}
    selectedBucketIndex={}
/>
```

Reply:
I wasn't totally sure if the logic to get the selected bucket was the responsibility of TransactionDistribution or TransactionOverview. But I totally agree with you that having no render effect is better.